src/main/src/main/java/org/geoserver/catalog/CoverageViewPamResourceInfo.java

```diff
@@ -0,0 +1,98 @@
+/* (c) 2024 Open Source Geospatial Foundation - all rights reserved
+ * This code is licensed under the GPL 2.0 license, available at the root
+ * application directory.
+ */
+package org.geoserver.catalog;
+
+import it.geosolutions.imageio.pam.PAMDataset;
+import java.io.IOException;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import org.geotools.api.coverage.grid.GridCoverageReader;
+import org.geotools.api.data.ResourceInfo;
+import org.geotools.coverage.grid.io.GridCoverage2DReader;
+import org.geotools.coverage.grid.io.PAMResourceInfo;
+import org.geotools.data.DefaultResourceInfo;
+
+/** A ResourceInfo wrapper for CoverageViews that provides access to the PAMDataset */
+public class CoverageViewPamResourceInfo extends DefaultResourceInfo implements PAMResourceInfo {
+    private final PAMDataset viewPam;
+
+    /**
+     * Constructor that populates the PAM dataset from the view bands
+     *
+     * @param inputAlphaNonNullCoverages input alpha non-null coverages
+     */
+    public CoverageViewPamResourceInfo(
+            CoverageViewReader.InputAlphaNonNullCoverages inputAlphaNonNullCoverages) {
+        viewPam = new PAMDataset();
+        populatePAMFromViewBands(inputAlphaNonNullCoverages, viewPam);
+    }
+
+    @Override
+    public PAMDataset getPAMDataset() {
+        return viewPam;
+    }
+
+    @Override
+    public boolean reloadPAMDataset() throws IOException {
+        return PAMResourceInfo.super.reloadPAMDataset();
+    }
+
+    /**
+     * Populates the PAM dataset from the view bands
+     *
+     * @param info convenience object to access the coverage view input readers
+     * @param viewPam the PAM dataset to populate
+     */
+    private void populatePAMFromViewBands(
+            CoverageViewReader.InputAlphaNonNullCoverages info, PAMDataset viewPam) {
+        Map<String, GridCoverageReader> readers = info.getInputReaders();
+        // iterate over the view bands and populate the PAM dataset
+        for (CoverageView.CoverageBand band : info.getBands()) {
+            for (CoverageView.InputCoverageBand inputCoverageBand : band.getInputCoverageBands()) {
+                // get the reader for the coverage associated with this input coverage band
+                GridCoverageReader reader1 = readers.get(inputCoverageBand.getCoverageName());
```

Comment:
Don't see the need for the "1" suffix.

```suggestion
                GridCoverageReader reader = readers.get(inputCoverageBand.getCoverageName());
```

Reply:
removed